c++/src/TypeImpl.cc

```diff
@@ -258,31 +258,34 @@ namespace orc {
 
     case STRUCT: {
       StructVectorBatch *result = new StructVectorBatch(capacity, memoryPool);
+      std::unique_ptr<ColumnVectorBatch> return_value = std::unique_ptr<ColumnVectorBatch>(result);
```

Comment:
maybe we can directly merge line 260 and 261 to a single statement. same for below.

Reply:
we need `result` of type `StructVectorBatch` to access fields on line 263